## 1 задача
Скачать файл в уроке

Считать данные с помощью pandas

Вывести на экран первые 5 строк

Посмотреть на описание признаков и на их содержание

In [2]:
import pandas as pd

In [12]:
df = pd.read_csv('kc-house-data.csv')
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [13]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

## 2 задача
Проведите первичный анализ данных

Изучите типы данных

Найдите количество пропущенных ячеек в данных

Посчитайте основные статистики по всем признакам и поизучайте их

Пишите выводы

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [15]:
df.dtypes

id                 int64
date              object
price            float64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront         int64
view               int64
condition          int64
grade              int64
sqft_above         int64
sqft_basement      int64
yr_built           int64
yr_renovated       int64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object

In [16]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

_Выводы:_ По результатам df.info видно, что пропущенных ячеек в таблице нет. dtypes показала, что большинство типов данных в таблице int, несколько колонок с float-значениями и один столбик с типом данных object - это дата. df.describe позволила вывести основные статистики, такие как среднее, стандартное отклонение, минимальное и максимальное значения, а также  25-й, 50-й и 75-й квартили. Так, можно увидеть, что средняя цена квартиры 5.400881e+05 и в среднем дома имеют один или два этажа (1,49). 

## 3 задача
### 3.1 В каком диапазоне изменяются стоимости недвижимости?
Стоимости недвижимости изменяются в диапазоне от 7.500000e+04 до 7.700000e+06

### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
Для нахождения средней доли жилой площади от всей площади дома:
${2079.899736\over1.510697e+04} \approx 0.14$

Жилая площадь занимает в среднем 14%.

### 3.3 Как много домов с разными этажами в данных?


In [18]:
df[df['floors'] > 1]

id             date     price  bedrooms  bathrooms  \
1      6414100192  20141209T000000  538000.0         3       2.25   
6      1321400060  20140627T000000  257500.0         3       2.25   
9      3793500160  20150312T000000  323000.0         3       2.50   
12      114101516  20140528T000000  310000.0         3       1.00   
14     1175000570  20150312T000000  530000.0         5       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
1             2570      7242     2.0           0     0  ...      7   
6             1715      6819     2.0           0     0  ...      7   
9             1890      6560     2.0           0     0  ...      7   
12            1430     19901     1.5           0     0  ...      7   
14            1810      4850     1.5           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
1            2170            400      1951          1991    98125  47.7210   
6            1715              0      1995             0    98003  47.3097   
9            1890              0      2003             0    98038  47.3684   
12           1430              0      1927             0    98028  47.7558   
14           1810              0      1900             0    98107  47.6700   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
1     -122.319           1690        7639  
6     -122.327           2238        6819  
9     -122.031           2390        7570  
12    -122.229           1780       12697  
14    -122.394           1360        4850  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[10933 rows x 21 columns]

В данных 10933 дома с несколькими этажами.

### 3.4 Насколько хорошие состояния у домов в данных?
Состояние домов оценивается по шкале от 0 до 5. Средний показатель равен 3,4. Большая часть домов имеет оценки 4 и 5. У большинства домов хорошее состояние. 

### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?
Первый дом был построен в 1900-м году, посдледний - в 2015-м.

## 4 задача
### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [23]:
df[df['floors'] == 2].describe()

id         price     bedrooms    bathrooms   sqft_living  \
count  8.241000e+03  8.241000e+03  8241.000000  8241.000000   8241.000000   
mean   4.677829e+09  6.488912e+05     3.631234     2.622619   2571.279092   
std    2.915394e+09  4.339377e+05     0.848208     0.621065    954.709667   
min    1.000102e+06  9.000000e+04     0.000000     0.000000    550.000000   
25%    2.180001e+09  3.788000e+05     3.000000     2.500000   1890.000000   
50%    3.982700e+09  5.429500e+05     4.000000     2.500000   2440.000000   
75%    7.518503e+09  7.725000e+05     4.000000     2.750000   3090.000000   
max    9.839301e+09  7.062500e+06    11.000000     7.750000  10040.000000   

           sqft_lot  floors   waterfront         view    condition  \
count  8.241000e+03  8241.0  8241.000000  8241.000000  8241.000000   
mean   1.613374e+04     2.0     0.009101     0.248149     3.187841   
std    4.462930e+04     0.0     0.094969     0.794362     0.469613   
min    5.720000e+02     2.0     0.000000     0.000000     1.000000   
25%    4.560000e+03     2.0     0.000000     0.000000     3.000000   
50%    7.089000e+03     2.0     0.000000     0.000000     3.000000   
75%    1.081000e+04     2.0     0.000000     0.000000     3.000000   
max    1.024068e+06     2.0     1.000000     4.000000     5.000000   

             grade   sqft_above  sqft_basement     yr_built  yr_renovated  \
count  8241.000000  8241.000000    8241.000000  8241.000000   8241.000000   
mean      8.360272  2398.800631     172.478461  1990.720908     86.009586   
std       1.153370   856.785751     369.923620    24.047390    405.412231   
min       5.000000   550.000000       0.000000  1900.000000      0.000000   
25%       8.000000  1810.000000       0.000000  1986.000000      0.000000   
50%       8.000000  2311.000000       0.000000  1998.000000      0.000000   
75%       9.000000  2880.000000       0.000000  2006.000000      0.000000   
max      13.000000  8860.000000    3260.000000  2015.000000   2015.000000   

            zipcode          lat         long  sqft_living15     sqft_lot15  
count   8241.000000  8241.000000  8241.000000    8241.000000    8241.000000  
mean   98065.350443    47.550779  -122.163837    2330.476398   13575.054969  
std       46.029378     0.135372     0.145675     743.000119   30929.111255  
min    98001.000000    47.155900  -122.519000     399.000000     651.000000  
25%    98030.000000    47.459000  -122.289000    1760.000000    4750.000000  
50%    98055.000000    47.564100  -122.168000    2260.000000    7113.000000  
75%    98103.000000    47.659300  -122.047000    2790.000000   10200.000000  
max    98199.000000    47.777600  -121.359000    6210.000000  871200.000000

Средняя стоимость дома с двумя спальнями = 6.488912e+05

### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [24]:
df[df['price'] > 6000000].describe()

id         price  bedrooms  bathrooms   sqft_living  \
count  3.000000e+00  3.000000e+00  3.000000   3.000000      3.000000   
mean   8.593434e+09  7.215833e+06  5.666667   6.750000  10660.000000   
std    1.613577e+09  4.285902e+05  0.577350   1.952562   1206.109448   
min    6.762700e+09  6.885000e+06  5.000000   4.500000   9890.000000   
25%    7.985800e+09  6.973750e+06  5.500000   6.125000   9965.000000   
50%    9.208900e+09  7.062500e+06  6.000000   7.750000  10040.000000   
75%    9.508800e+09  7.381250e+06  6.000000   7.875000  11045.000000   
max    9.808701e+09  7.700000e+06  6.000000   8.000000  12050.000000   

           sqft_lot    floors  waterfront  view  condition      grade  \
count      3.000000  3.000000    3.000000   3.0   3.000000   3.000000   
mean   32099.666667  2.166667    0.333333   3.0   3.333333  12.333333   
std     4902.943028  0.288675    0.577350   1.0   0.577350   1.154701   
min    27600.000000  2.000000    0.000000   2.0   3.000000  11.000000   
25%    29487.000000  2.000000    0.000000   2.5   3.000000  12.000000   
50%    31374.000000  2.000000    0.000000   3.0   3.000000  13.000000   
75%    34349.500000  2.250000    0.500000   3.5   3.500000  13.000000   
max    37325.000000  2.500000    1.000000   4.0   4.000000  13.000000   

        sqft_above  sqft_basement     yr_built  yr_renovated       zipcode  \
count     3.000000       3.000000     3.000000      3.000000      3.000000   
mean   8370.000000    2290.000000  1950.333333   1329.333333  98048.333333   
std     614.898366    1226.499083    46.371687   1151.257718     49.662192   
min    7680.000000    1030.000000  1910.000000      0.000000  98004.000000   
25%    8125.000000    1695.000000  1925.000000    993.500000  98021.500000   
50%    8570.000000    2360.000000  1940.000000   1987.000000  98039.000000   
75%    8715.000000    2920.000000  1970.500000   1994.000000  98070.500000   
max    8860.000000    3480.000000  2001.000000   2001.000000  98102.000000   

             lat       long  sqft_living15    sqft_lot15  
count   3.000000    3.00000       3.000000      3.000000  
mean   47.636767 -122.25900    4136.666667  25659.666667  
std     0.011466    0.05693     349.332697  16965.980972  
min    47.629800 -122.32300    3930.000000   8800.000000  
25%    47.630150 -122.28150    3935.000000  17124.500000  
50%    47.630500 -122.24000    3940.000000  25449.000000  
75%    47.640250 -122.22700    4240.000000  34089.500000  
max    47.650000 -122.21400    4540.000000  42730.000000

Средняя общая площадь таких домов = 32099.666667

### 4.3 Как много домов коснулся ремонт?

In [27]:
df[df['yr_renovated'] > 0].describe()

id         price    bedrooms   bathrooms   sqft_living  \
count  9.140000e+02  9.140000e+02  914.000000  914.000000    914.000000   
mean   4.348365e+09  7.603790e+05    3.452954    2.298961   2320.693654   
std    2.893205e+09  6.077348e+05    1.040514    0.890498   1083.822532   
min    3.600057e+06  1.100000e+05    1.000000    0.750000    520.000000   
25%    1.863253e+09  4.012500e+05    3.000000    1.750000   1554.750000   
50%    3.699650e+09  6.000000e+05    3.000000    2.250000   2200.000000   
75%    6.918700e+09  9.000000e+05    4.000000    2.750000   2877.500000   
max    9.829200e+09  7.700000e+06   11.000000    8.000000  12050.000000   

            sqft_lot      floors  waterfront        view   condition  \
count     914.000000  914.000000  914.000000  914.000000  914.000000   
mean    16633.631291    1.510394    0.045952    0.613786    3.223195   
std     38024.835725    0.494937    0.209495    1.219891    0.519571   
min      1024.000000    1.000000    0.000000    0.000000    1.000000   
25%      5000.000000    1.000000    0.000000    0.000000    3.000000   
50%      7500.000000    1.500000    0.000000    0.000000    3.000000   
75%     13064.500000    2.000000    0.000000    0.000000    3.000000   
max    478288.000000    3.000000    1.000000    4.000000    5.000000   

            grade   sqft_above  sqft_basement     yr_built  yr_renovated  \
count  914.000000   914.000000     914.000000   914.000000    914.000000   
mean     7.735230  1879.728665     440.964989  1939.527352   1995.827133   
std      1.205103   848.591483     542.239477    22.621409     15.517107   
min      4.000000   520.000000       0.000000  1900.000000   1934.000000   
25%      7.000000  1270.000000       0.000000  1920.250000   1987.000000   
50%      8.000000  1745.000000     145.000000  1942.000000   2000.000000   
75%      8.000000  2290.000000     820.000000  1955.000000   2007.000000   
max     13.000000  8570.000000    4820.000000  2003.000000   2015.000000   

            zipcode         lat        long  sqft_living15     sqft_lot15  
count    914.000000  914.000000  914.000000     914.000000     914.000000  
mean   98094.320569   47.579405 -122.259721    1977.565646   13797.580963  
std       54.687393    0.122513    0.131604     704.283165   26870.455079  
min    98001.000000   47.196000 -122.511000     828.000000    1281.000000  
25%    98040.000000   47.516425 -122.356000    1460.000000    5040.000000  
50%    98112.000000   47.600050 -122.291000    1835.000000    7500.000000  
75%    98126.000000   47.671775 -122.206250    2350.000000   11367.750000  
max    98199.000000   47.777000 -121.359000    5500.000000  310582.000000

Ремонт коснулся 9.14000000 домов

### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [28]:
df[df['grade'] > 10].describe()

id         price    bedrooms   bathrooms   sqft_living  \
count  5.020000e+02  5.020000e+02  502.000000  502.000000    502.000000   
mean   4.115276e+09  1.678635e+06    4.189243    3.628984   4668.340637   
std    2.756606e+09  9.197760e+05    0.846529    0.951373   1293.931170   
min    4.610050e+07  4.200000e+05    0.000000    0.000000   2370.000000   
25%    1.853000e+09  1.075000e+06    4.000000    3.250000   3870.000000   
50%    3.624275e+09  1.407500e+06    4.000000    3.500000   4400.000000   
75%    6.725376e+09  1.950000e+06    5.000000    4.000000   5187.500000   
max    9.831200e+09  7.700000e+06    9.000000    8.000000  13540.000000   

            sqft_lot      floors  waterfront        view   condition  \
count     502.000000  502.000000  502.000000  502.000000  502.000000   
mean    41004.627490    1.959163    0.061753    1.157371    3.197211   
std     92815.766519    0.391815    0.240946    1.495547    0.463166   
min       609.000000    1.000000    0.000000    0.000000    3.000000   
25%     11602.500000    2.000000    0.000000    0.000000    3.000000   
50%     16907.000000    2.000000    0.000000    0.000000    3.000000   
75%     34172.500000    2.000000    0.000000    2.000000    3.000000   
max    920423.000000    3.500000    1.000000    4.000000    5.000000   

            grade   sqft_above  sqft_basement     yr_built  yr_renovated  \
count  502.000000   502.000000     502.000000   502.000000    502.000000   
mean    11.231076  4053.081673     615.258964  1991.784861     83.531873   
std      0.479511  1094.621372     817.804651    20.645993    400.177892   
min     11.000000  1910.000000       0.000000  1902.000000      0.000000   
25%     11.000000  3360.000000       0.000000  1989.000000      0.000000   
50%     11.000000  3990.000000       0.000000  1997.000000      0.000000   
75%     11.000000  4567.500000    1127.500000  2003.000000      0.000000   
max     13.000000  9410.000000    4820.000000  2015.000000   2010.000000   

            zipcode         lat        long  sqft_living15     sqft_lot15  
count    502.000000  502.000000  502.000000     502.000000     502.000000  
mean   98054.278884   47.600295 -122.152239    3621.922311   29719.916335  
std       46.219837    0.084725    0.123246     814.288977   65465.234993  
min    98003.000000   47.290900 -122.416000    1280.000000    1229.000000  
25%    98014.000000   47.550075 -122.224750    3120.000000   11063.000000  
50%    98041.000000   47.602450 -122.146500    3640.000000   15136.000000  
75%    98075.000000   47.649700 -122.072250    4117.500000   25448.500000  
max    98199.000000   47.771100 -121.756000    6210.000000  871200.000000

In [29]:
df[df['grade'] < 4].describe()

id          price  bedrooms  bathrooms  sqft_living  \
count  4.000000e+00       4.000000      4.00     4.0000     4.000000   
mean   3.880987e+09  189750.000000      0.75     0.1875   520.000000   
std    1.375205e+09   98001.275502      0.50     0.3750   165.126214   
min    2.420069e+09   75000.000000      0.00     0.0000   290.000000   
25%    3.170827e+09  125250.000000      0.75     0.0000   462.500000   
50%    3.700690e+09  202000.000000      1.00     0.0000   560.000000   
75%    4.410850e+09  266500.000000      1.00     0.1875   617.500000   
max    5.702500e+09  280000.000000      1.00     0.7500   670.000000   

           sqft_lot  floors  waterfront  view  condition  grade  sqft_above  \
count      4.000000     4.0         4.0   4.0   4.000000    4.0    4.000000   
mean   25433.500000     1.0         0.0   0.0   2.750000    2.5  520.000000   
std    12892.936399     0.0         0.0   0.0   1.707825    1.0  165.126214   
min    12981.000000     1.0         0.0   0.0   1.000000    1.0  290.000000   
25%    18901.500000     1.0         0.0   0.0   1.750000    2.5  462.500000   
50%    22688.000000     1.0         0.0   0.0   2.500000    3.0  560.000000   
75%    29220.000000     1.0         0.0   0.0   3.500000    3.0  617.500000   
max    43377.000000     1.0         0.0   0.0   5.000000    3.0  670.000000   

       sqft_basement     yr_built  yr_renovated       zipcode        lat  \
count            4.0     4.000000           4.0      4.000000   4.000000   
mean             0.0  1949.750000           0.0  98028.250000  47.383600   
std              0.0    21.013884           0.0     11.206397   0.171939   
min              0.0  1920.000000           0.0  98022.000000  47.208200   
25%              0.0  1942.500000           0.0  98022.000000  47.249900   
50%              0.0  1956.500000           0.0  98023.000000  47.397300   
75%              0.0  1963.750000           0.0  98029.250000  47.531000   
max              0.0  1966.000000           0.0  98045.000000  47.531600   

             long  sqft_living15    sqft_lot15  
count    4.000000       4.000000      4.000000  
mean  -121.884500    1277.500000  25128.500000  
std      0.101727     269.366541  12823.494908  
min   -121.995000     990.000000  12233.000000  
25%   -121.928250    1117.500000  19970.000000  
50%   -121.897000    1250.000000  22699.500000  
75%   -121.853250    1410.000000  27858.000000  
max   -121.749000    1620.000000  42882.000000

Стоимость этих домов отличается на: $1.678635e+06 - 189750.000000 = 1488885$

## 5 задача
### 5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [32]:
df[(df['waterfront'] > 0) & (df['bathrooms'] >= 3) & (df['sqft_basement'] > 0)]

id             date      price  bedrooms  bathrooms  \
300    3225069065  20140624T000000  3075000.0         4       5.00   
457    2122039094  20141126T000000   705000.0         3       3.00   
1164   1247600105  20141020T000000  5110800.0         5       5.25   
1261   7636800041  20140625T000000   995000.0         3       4.50   
1315   7558700030  20150413T000000  5300000.0         6       6.00   
1361   1925069082  20150511T000000  2200000.0         5       4.25   
2626   7738500731  20140815T000000  4500000.0         5       5.50   
2864   4114601570  20141118T000000  3600000.0         3       3.25   
3914   9808700762  20140611T000000  7062500.0         5       4.50   
3934   1269200229  20140723T000000  1379900.0         3       3.25   
5760    623059016  20140717T000000  1100000.0         4       3.25   
6233   2024069008  20140619T000000  2200000.0         5       4.75   
6508   4217402115  20150421T000000  3650000.0         6       4.75   
7989   9362000040  20140623T000000  3400000.0         3       4.50   
8092   1924059029  20140617T000000  4668000.0         5       6.75   
8163   1724069079  20150319T000000  1452000.0         2       3.25   
8784   3343302110  20150306T000000  1800000.0         3       3.00   
10263   625059036  20140813T000000  2700000.0         5       4.00   
11257  3761100045  20140618T000000  3000000.0         4       4.25   
11402  3426049284  20140819T000000  2300000.0         4       3.25   
11424  5119400075  20140620T000000   950000.0         3       3.25   
11446   624069035  20141209T000000  2750000.0         4       4.00   
11844  2623039082  20150218T000000   770000.0         3       3.50   
12459  4114601580  20140724T000000  1900000.0         6       4.00   
12935  3222049044  20140612T000000   835000.0         3       3.00   
13710  2923500230  20141216T000000  2600000.0         4       4.50   
13967  3126059023  20150303T000000  3395000.0         4       3.50   
14083  3625059043  20140904T000000  3300000.0         5       4.75   
14619  7936500109  20140725T000000  2230000.0         3       3.00   
14822  3623500408  20150330T000000  2600000.0         3       3.00   
15482   624069108  20140812T000000  3200000.0         4       3.25   
15668  4030100290  20141001T000000  1680000.0         5       3.50   
16817   825059178  20140923T000000  2574000.0         4       3.75   
17767  7352200450  20150115T000000  2050000.0         4       3.25   
18199  3625059152  20141230T000000  3300000.0         3       3.25   
18455  8043700300  20140608T000000  2700000.0         4       3.25   
18556  6613000930  20140902T000000  2950000.0         4       3.25   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
300           4550     18641     1.0           1     4  ...     10   
457           1970     20978     2.0           1     3  ...      9   
1164          8010     45517     2.0           1     4  ...     12   
1261          4380     47044     2.0           1     3  ...      9   
1315          7390     24829     2.0           1     4  ...     12   
1361          4640     22703     2.0           1     4  ...      8   
2626          6640     40014     2.0           1     4  ...     12   
2864          5020     12431     2.0           1     4  ...     10   
3914         10040     37325     2.0           1     2  ...     11   
3934          3786     38038     1.0           1     4  ...      9   
5760          3190     11774     2.0           1     4  ...      8   
6233          5990     10450     2.0           1     4  ...     11   
6508          5480     19401     1.5           1     4  ...     11   
7989          5230     17826     2.0           1     4  ...     10   
8092          9640     1

У клиента есть 41 вариант.

### 5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [33]:
df[(df['view'] == 4) | ((df['waterfront'] > 0) & (df['condition'] == 5) & (df['yr_built'] >= 1980))].describe()

id         price    bedrooms   bathrooms   sqft_living  \
count  3.190000e+02  3.190000e+02  319.000000  319.000000    319.000000   
mean   4.991007e+09  1.463711e+06    3.633229    2.793887   3351.473354   
std    2.931542e+09  9.522096e+05    1.107561    1.079593   1553.260668   
min    9.800107e+07  2.520000e+05    1.000000    0.750000    680.000000   
25%    2.608125e+09  8.175000e+05    3.000000    2.250000   2425.000000   
50%    4.114602e+09  1.185000e+06    4.000000    2.500000   3050.000000   
75%    7.855100e+09  1.850000e+06    4.000000    3.250000   3911.000000   
max    9.829200e+09  6.885000e+06    7.000000    8.000000  13540.000000   

            sqft_lot      floors  waterfront   view   condition       grade  \
count     319.000000  319.000000  319.000000  319.0  319.000000  319.000000   
mean    21594.987461    1.605016    0.423197    4.0    3.567398    9.078370   
std     35742.685384    0.580696    0.494842    0.0    0.744584    1.604868   
min      1171.000000    1.000000    0.000000    4.0    1.000000    5.000000   
25%      8421.000000    1.000000    0.000000    4.0    3.000000    8.000000   
50%     13122.000000    1.500000    0.000000    4.0    3.000000    9.000000   
75%     20712.000000    2.000000    1.000000    4.0    4.000000   10.000000   
max    322188.000000    3.000000    1.000000    4.0    5.000000   13.000000   

        sqft_above  sqft_basement     yr_built  yr_renovated       zipcode  \
count   319.000000     319.000000   319.000000    319.000000    319.000000   
mean   2516.529781     834.943574  1966.150470    349.623824  98095.752351   
std    1242.594566     768.668466    27.219165    758.888162     61.667332   
min     680.000000       0.000000  1903.000000      0.000000  98001.000000   
25%    1680.000000       0.000000  1948.000000      0.000000  98036.000000   
50%    2180.000000     810.000000  1968.000000      0.000000  98115.000000   
75%    2940.000000    1375.000000  1989.000000      0.000000  98146.000000   
max    9410.000000    4820.000000  2014.000000   2015.000000  98198.000000   

              lat        long  sqft_living15     sqft_lot15  
count  319.000000  319.000000     319.000000     319.000000  
mean    47.575866 -122.263708    2843.752351   18867.141066  
std      0.112742    0.131633     775.314206   27496.960486  
min     47.249800 -122.511000    1010.000000    1505.000000  
25%     47.513300 -122.376000    2385.000000    8311.500000  
50%     47.578400 -122.276000    2810.000000   12528.000000  
75%     47.659800 -122.194000    3290.000000   19014.500000  
max     47.775700 -121.739000    5380.000000  248292.000000

Стоимость таких домов будет варьироваться в пределах от 2.52000000000 до 6.885000000000	

### 5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [35]:
df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)].describe()

id          price  bedrooms  bathrooms  sqft_living  \
count  6.000000e+00       6.000000  6.000000   6.000000     6.000000   
mean   6.508735e+09  128916.666667  2.666667   1.375000  1143.333333   
std    1.896467e+09   19489.527102  0.516398   0.306186   129.099445   
min    4.239400e+09   90000.000000  2.000000   1.000000   980.000000   
25%    5.162553e+09  131000.000000  2.250000   1.125000  1035.000000   
50%    6.432705e+09  136750.000000  3.000000   1.500000  1200.000000   
75%    7.560350e+09  139875.000000  3.000000   1.500000  1222.500000   
max    9.272202e+09  140000.000000  3.000000   1.750000  1270.000000   

          sqft_lot  floors  waterfront  view  condition     grade  \
count     6.000000     6.0         6.0   6.0   6.000000  6.000000   
mean   4507.333333     2.0         0.0   0.0   2.833333  7.000000   
std    3019.489471     0.0         0.0   0.0   0.983192  0.632456   
min    1561.000000     2.0         0.0   0.0   1.000000  6.000000   
25%    2124.000000     2.0         0.0   0.0   3.000000  7.000000   
50%    3745.000000     2.0         0.0   0.0   3.000000  7.000000   
75%    6500.000000     2.0         0.0   0.0   3.000000  7.000000   
max    8991.000000     2.0         0.0   0.0   4.000000  8.000000   

        sqft_above  sqft_basement     yr_built  yr_renovated       zipcode  \
count     6.000000            6.0     6.000000      6.000000      6.000000   
mean   1143.333333            0.0  1954.833333    333.833333  98060.166667   
std     129.099445            0.0    31.858542    817.721326     37.279574   
min     980.000000            0.0  1908.000000      0.000000  98014.000000   
25%    1035.000000            0.0  1933.000000      0.000000  98042.000000   
50%    1200.000000            0.0  1967.500000      0.000000  98048.500000   
75%    1222.500000            0.0  1978.000000      0.000000  98082.750000   
max    1270.000000            0.0  1983.000000   2003.000000  98116.000000   

             lat        long  sqft_living15   sqft_lot15  
count   6.000000    6.000000        6.00000     6.000000  
mean   47.468650 -122.068667     1585.00000  4539.666667  
std     0.151972    0.358690      868.80953  2706.158803  
min    47.317000 -122.384000      980.00000  1848.000000  
25%    47.361425 -122.187250     1097.50000  2470.750000  
50%    47.421350 -122.165500     1270.00000  4077.000000  
75%    47.557700 -122.149000     1562.50000  5750.000000  
max    47.707600 -121.359000     3290.00000  8993.000000

Средняя оценка по состоянию у таких домов = 2,83.